webapp/src/components/CreateMeetingNoteForm.tsx

```diff
@@ -0,0 +1,112 @@
+import { Alert, Card, CardContent, Snackbar, TextField } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import React, { FormEventHandler, useState, useContext } from 'react';
+
+import { EntityId } from '../types/api';
+import SessionContext from './SessionContext';
+
+interface CreateMeetingNoteFormProps {
+  meetingId: EntityId;
+  onMeetingNoteChanged?: (id: EntityId) => void;
+  greatestSortOrderOfMeetingNotesList: number;
+}
+
+const CreateMeetingNoteForm = ({
+  meetingId,
+  onMeetingNoteChanged,
+  greatestSortOrderOfMeetingNotesList,
+}: CreateMeetingNoteFormProps) => {
+  const { principalId } = useContext(SessionContext);
+
+  const [isSavingMeetingNote, setIsSavingMeetingNote] = useState(false);
+  const [saveMeetingNoteError, setSaveMeetingNoteError] = useState(null);
+  const [isSuccessMessageVisible, setIsSuccessMessageVisible] = useState(false);
+  const [meetingNoteText, setMeetingNoteText] = useState('');
+  const onSubmit: FormEventHandler = event => {
+    event.preventDefault();
+    setIsSavingMeetingNote(true);
+
+    fetch(`${process.env.REACT_APP_API_ORIGIN}/meetings/${meetingId}/notes`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        agenda_owning_participant_id: principalId,
```

Comment:
Why is a principal ID being set as a participant ID? Those are different kinds of objects, and they have different ID sequences.

Reply:
I don't understand this response. The system doesn't have "users". @pthan1: If by "user" you mean "principal", then that's the same as the current, incorrect code. I'm also unsure what you mean by "is the objet type to use" since both principal and participant ids are the same type: `EntityId`.

Reply:
I saw a Participant type that I thought could be used.  However, looking at the `Meeting` component, there is a `currentParticipantId` constant, should I pass that down as props to the form to use to set the participant ID for the POST request?

Reply:
Sounds reasonable.